src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -1407,5 +1427,272 @@ End Class
 End Namespace";
             await VerifyVB.VerifyAnalyzerAsync(code);
         }
+
+        [Fact]
+        public async Task TestEnumeratedParameterAfterLinqCallChain()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> i, IEnumerable<int> j, IEnumerable<int> k)
+    {
+        var z = [|k|].Concat([|j|]).Concat(i);
```

Comment:
we can report the diagnostic here,
we could also report multiple diagnostics (`k`, and `j`) on `[|z|].ToArray()`

Tag @mikadumont to verify the behavior here